pysat/tests/instrument_tests.py

```diff
@@ -0,0 +1,239 @@
+"""
+tests the pysat instruments and code
+"""
+import datetime as dt
+from importlib import import_module
+import os
+import tempfile
+import warnings
+
+import pytest
+
+import pysat
+
+# dict, keyed by pysat instrument, with a list of usernames and passwords
+user_download_dict = {'supermag_magnetometer': {'user': 'rstoneback',
+                                                'password': 'None'}}
+
+def remove_files(inst):
+    # remove any files downloaded as part of the unit tests
+    temp_dir = inst.files.data_path
+    # Check if there are less than 20 files to ensure this is the testing
+    # directory
+    if len(inst.files.files.values) < 20:
+        for the_file in list(inst.files.files.values):
+            # Check if filename is appended with date for fake_daily data
+            # ie, does an underscore exist to the right of the file extension?
+            if the_file.rfind('_') > the_file.rfind('.'):
+                # If so, trim the appendix to get the original filename
+                the_file = the_file[:the_file.rfind('_')]
+            file_path = os.path.join(temp_dir, the_file)
+            if os.path.isfile(file_path):
+                os.unlink(file_path)
+    else:
+        warnings.warn(''.join(('Files > 20.  Not deleted.  Please check to ',
+                               'ensure temp directory is used')))
+
+
+def generate_instrument_list(instrument_names=[], package=None):
+    """Iterate through and create all of the test Instruments needed.
+       Only want to do this once.
+
+    """
+
+    if package == None:
+        package = 'pysat.instruments'
+
+    instrument_download = []
+    instrument_no_download = []
+
+    # create temporary directory
+    dir_name = tempfile.mkdtemp()
+    saved_path = pysat.data_dir
+    pysat.utils.set_data_dir(dir_name, store=False)
+
+    for name in instrument_names:
+        try:
+            module = import_module(''.join(('.', name)),
+                                   package=package)
+        except ImportError:
+            print(' '.join(["Couldn't import", name]))
+            pass
+        else:
+            # try to grab basic information about the module so we
+            # can iterate over all of the options
+            try:
+                info = module._test_dates
+            except AttributeError:
+                info = {}
+                info[''] = {'': pysat.datetime(2009, 1, 1)}
```

Comment:
Was this intentional?  Since this is new seems no harm to use `dt.datetime`

Reply:
fixed.